glean-core/rlb/src/private/dynamic/timespan.rs

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::sync::atomic::{AtomicU64, Ordering::SeqCst};
+use std::sync::Arc;
+
+use glean_core::{ErrorType, metrics::TimeUnit};
+use inherent::inherent;
+
+use crate::{dispatcher, new_metric};
+
+/// Timespan metric wrapper around the FFI implementation
+#[derive(Clone)]
+pub struct TimespanMetric(Arc<AtomicU64>);
+
+impl TimespanMetric {
+    /// The public constructor used by automatically generated metrics.
+    pub fn new(meta: glean_core::CommonMetricData, time_unit: TimeUnit) -> Self {
+        let metric = Arc::new(AtomicU64::new(0));
+
+        let deferred = Arc::clone(&metric);
+        dispatcher::launch(move || {
+            let metric = new_metric!(glean_new_timespan_metric, meta, time_unit as i32);
+            deferred.store(metric, SeqCst);
```

Comment:
See commit message: c92e04140b3e84bdad5cb463d794dd2f638b13c8

Reply:
Oh, so everything will need to be two-step constructed: ctor then init. 

Okay, I think I've got it.